From: Jan Kara <jack@suse.com>
Subject: fs: Avoid deadlocks of fsync_bdev() and fs freezing
References: bsc#935123
Patch-mainline: Never, fixed differently by 5accdf82ba25a and company

Use get_super_thawed() in fsync_bdev() so that we get superblock that is
thawed and thus won't block on frozen filesystem.

Signed-off-by: Jan Kara <jack@suse.com>

---
 fs/block_dev.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/block_dev.c
+++ b/fs/block_dev.c
@@ -176,7 +176,7 @@ EXPORT_SYMBOL(sync_blockdev);
  */
 int fsync_bdev(struct block_device *bdev)
 {
-	struct super_block *sb = get_super(bdev);
+	struct super_block *sb = get_super_thawed(bdev);
 	if (sb) {
 		int res = sync_filesystem(sb);
 		drop_super(sb);
